fs/ignorefs/ignorefs.go

```diff
@@ -283,8 +286,22 @@ func (d *ignoreDirectory) buildContext(ctx context.Context) (*ignoreContext, err
 
 	for _, dotfile := range effectiveDotIgnoreFiles {
 		if e, err := d.Directory.Child(ctx, dotfile); err == nil {
-			if f, ok := e.(fs.File); ok {
-				dotIgnoreFiles = append(dotIgnoreFiles, f)
+			switch entry := e.(type) {
+			case fs.File:
+				dotIgnoreFiles = append(dotIgnoreFiles, entry)
+
+			case fs.Symlink:
+				target, err := entry.Resolve(ctx)
+				if err != nil {
+					link, _ := entry.Readlink(ctx)
+					return nil, errors.Wrapf(err, "when resolving symlink %s of type %T, which points to %s", entry.Name(), entry, link)
+				}
+
+				if f, ok := target.(fs.File); !ok {
```

Comment:
what about symlink->symlink->file, would it make sense to resolve symlinks in a loop until we encounter a `fs.File` ?